jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -0,0 +1,148 @@
+        ).values()
+        [uncert_cube] = self._app.get_data_from_viewer(
+            self._app._jdaviz_helper._default_uncert_viewer_reference_name,
+            include_subsets=False
+        ).values()
+
+        # This plugin collapses over the *spatial axes* (optionally over a spatial subset,
+        # defaults to ``No Subset``). Since the Cubeviz parser puts the fluxes
+        # and uncertainties in different glue Data objects, we translate the spectral
+        # cube and its uncertainties into separate NDDataArrays, then combine them:
+        if self.spatial_subset_selected != self.spatial_subset.default_text:
+            nddata = spectral_cube.get_subset_object(
+                subset_id=self.spatial_subset_selected, cls=NDDataArray
+            )
+            uncertainties = uncert_cube.get_subset_object(
+                subset_id=self.spatial_subset_selected, cls=NDDataArray
+            ).data
+        else:
+            nddata = spectral_cube.get_object(cls=NDDataArray)
+            uncertainties = uncert_cube.get_object(cls=NDDataArray).data
+
+        # We attach the uncertainties to the NDDataArray with the data:
+        nddata.uncertainty = StdDevUncertainty(uncertainties)
+
+        # Collapse an e.g. 3D spectral cube to 1D spectrum, assuming that last axis
+        # is always wavelength. This will need adjustment after the following
+        # specutils PR is merged: https://github.com/astropy/specutils/pull/999
+        spatial_axes = tuple(range(nddata.ndim - 1))
+
+        # by default we want to use operation_ignores_mask=True in nddata:
+        kwargs.setdefault("operation_ignores_mask", True)
+        # by default we want to propagate uncertainties:
+        kwargs.setdefault("propagate_uncertainties", True)
+
+        collapsed_spec = getattr(nddata, self.function_selected.lower())(
+            axis=spatial_axes, **kwargs
+        )  # returns an NDDataArray
+
+        # Hack to extract the spectral axis with units converted
+        # to the wavelength axis in specviz:
+        collapsed_spec.wcs = None
+        collapsed_spec.wcs = PaddedSpectrumWCS(
+            SlicedFITSWCS(spectral_cube.coords, spatial_axes), 1
+        )
+        self._set_default_results_label()
+
+        if add_data:
+            self.add_results.add_results_from_plugin(
+                collapsed_spec, label=self.results_label, replace=False
+            )
+
+            snackbar_message = SnackbarMessage(
+                "Spectrum extracted successfully.",
+                color="success",
+                sender=self)
+            self.hub.broadcast(snackbar_message)
+
+        return collapsed_spec
+
+    def vue_spectral_extraction(self, *args, **kwargs):
+        self.collapse_to_spectrum(add_data=True)
+
+    @observe("results_label")
+    def _set_default_results_label(self, event={}):
+        self.results_label_default = (
+            self._app.data_collection[0].label + " extracted"
+        )
```

Comment:
The way this is implemented now, the label will be set when clicking on the button, which is confusing for the user. 
 If this isn't going to depend on any other selection (subset or dataset for example), then having an observe seems like overkill.  I'd suggest putting it in `__init__`, but then there's a chance that `data_collection[0]` doesn't exist yet (which is probably a case that needs to be checked elsewhere, perhaps by disabling the whole plugin until data is loaded).  Alternatively, we could use a `DatasetSelect` component that only allows for flux cubes so it will only have one choice and observe that - then if we ever remove the restriction of a single cube, this plugin will still work.

Reply:
Solution implemented in 732faf4, which no longer copies the name of the dataset. The default label is now "Spectral extraction" and if you select a subset, it gets added in parens, like: "Spectral extraction (Subset 1)".